src/inference/src/dev/icompiled_model.cpp

```diff
@@ -64,6 +65,18 @@ ov::ICompiledModel::ICompiledModel(const std::shared_ptr<const ov::Model>& model
             new_param->set_element_type(param->get_element_type());
             new_param->set_layout(param->get_layout());
             new_param->output(0).get_rt_info() = param->output(0).get_rt_info();
+            auto old_tensor = param->output(0).get_tensor_ptr();
+            if (tensor_map.count(old_tensor)) {
+                new_param->output(0).set_tensor_ptr(tensor_map[old_tensor]);
+            } else {
+                auto tensor_desc =
+                    std::make_shared<ov::descriptor::Tensor>(param->get_output_element_type(0),
+                                                             param->get_output_partial_shape(0),
+                                                             new_param->output(0).get_tensor().get_names());
+                tensor_desc->clone_from(param->output(0).get_tensor());
+                new_param->output(0).set_tensor_ptr(tensor_desc);
+                tensor_map[old_tensor] = tensor_desc;
```

Comment:
Why do you need set_tensor here?
I thought that you can simple call:
```
tensor_map[old_tensor] = new_param->output(0).get_tensor_ptr();
```

Reply:
sure, it is reasonable, let me try whether there are any test failures.